corehq/apps/userreports/views.py

```diff
@@ -515,14 +521,22 @@ def _get_config_kwargs(self, app_source):
             )
         }
 
-    def _build_data_source(self, app_source, username):
+    def _expire_data_source(self, data_source_config_id):
+        with temp_setting_value('CELERY_ALWAYS_EAGER', False):  # so it works in dev environments too
```

Comment:
Almost. It executes it immediately. So as soon as the data source is created, it gets deleted, and you've got no preview.